Selenium task

olgazhuck opened the pull request:
Selenium homework

olgazhuck committed:
Selenium task

eugene-okulik commented on homework/volha_zhuck/lesson24/test_Selenium_1.py:
```diff
@@ -0,0 +1,114 @@
+    text_email.send_keys(email)
+    text_email = driver.find_element(By.ID, 'userNumber')
+    text_email.send_keys(phone)
+    text_subject = driver.find_element(By.ID, 'subjectsInput')
+    text_subject.send_keys(subject)
```
В конце заполнения всей формы это поле (subject) остается пустым. Тесты - это тоже программа, а любую программу нужно тестировать и перед завершением работы с тестами нужно убеждаться, что эта программа работает правильно

olgazhuck committed:
- оптимизировала тест
-добавила прокрутку страницы, ибо сегодня из-за рекламы тест не работал вообще, возможно можно было как-то по-другому это реализовать
-добавила ассерт, который отлавливает баг на сайте

eugene-okulik commented on homework/volha_zhuck/lesson24/test_Selenium_1.py:
```diff
@@ -0,0 +1,127 @@
+    date_element = driver.find_element(
+        By.XPATH, "//div[@aria-label='Choose Tuesday, February 20th, 1990']"
+    )
+    date_element.click()
+    driver.find_element(By.CLASS_NAME, 'css-19bqh2r').click()
+    driver.execute_script("window.scrollBy(0, 300);")
+    button = driver.find_element(By.ID, 'submit')
+    button.click()
+    table = driver.find_element(By.CSS_SELECTOR, "table.table")
+    rows = table.find_elements(By.CSS_SELECTOR, "tbody tr")
+    headers = table.find_elements(By.CSS_SELECTOR, "thead th")
+    header_text = [header.text for header in headers]
+    print(" | ".join(header_text))
+
+    for row in rows:
+        cells = row.find_elements(By.TAG_NAME, "td")
+        cell_text = [cell.text for cell in cells]
+        print(" | ".join(cell_text))
+
+    expected_table_data = {
+        "Student Name": f"{form_data['firstName']} {form_data['lastName']}",
+        "Student Email": form_data['userEmail'],
+        "Gender": "Male",
+        "Mobile": form_data['userNumber'],
+        "Date of Birth": "20 February,1990",
+        "Subjects": form_data['subjectsInput'],
+        "Hobbies": "",
+        "Picture": "",
+        "Address": form_data['currentAddress'],
+        "State and City": "Haryana Karnal"
+    }
+    for row in rows:
+        cells = row.find_elements(By.TAG_NAME, "td")
+        label = cells[0].text
+        value = cells[1].text
+        expected_value = expected_table_data[label]
+        assert value == expected_value, (
+            f"Error: Expected Values: '{expected_value}', "
+            f"Actual: '{value}' in Label: '{label}'"
+        )
```
Сейчас subjects и hobby не заполняются

olgazhuck committed:
- исправила subject и добавила хобби, тест стал проходить

olgazhuck committed:
исправлены ошибки
